test/gtest/uct/test_peer_failure.cc

```diff
@@ -461,16 +461,19 @@ UCT_INSTANTIATE_TEST_CASE(test_uct_peer_failure_multiple)
 class test_uct_keepalive : public uct_test {
 public:
     test_uct_keepalive() :
-        m_pid(getpid()), m_entity(NULL), m_err_handler_count(0)
+        m_ka(NULL), m_pid(getpid()), m_entity(NULL), m_err_handler_count(0)
     {
-        m_ka.start_time = 0;
     }
 
     void init()
     {
         m_entity = create_entity(0, err_handler_cb);
         m_entity->connect(0, *m_entity, 0);
         m_entities.push_back(m_entity);
+
+        ASSERT_TRUE(has_mm());
+        uct_mm_ep_t *ep = ucs_derived_of(m_entity->ep(0), uct_mm_ep_t);
+        m_ka            = &ep->keepalive;
```

Comment:
minor: i would add a getter function to return ka pointer instead of caching it in a member variable
these 3 lines could be the body of that getter function